app/src/main/java/com/example/habitsmasher/ui/dashboard/AddHabitDialog.java

```diff
@@ -147,120 +139,19 @@ public void onClick(View view) {
-                if(_tracker.getThursday()){
-                    _tracker.setThursday(false);
-                }
-                else{
-                    _tracker.setThursday(true);
-                }
-                Log.d("Tracker Status", _tracker.getDays());
-            }
-        });
-
-        _fridayButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                if(_tracker.getFriday()){
-                    _tracker.setFriday(false);
-                }
-                else{
-                    _tracker.setFriday(true);
-                }
-                Log.d("Tracker Status", _tracker.getDays());
-            }
-        });
+        _thursdayButton.setOnClickListener(new ClickListenerForDaysOfTheWeek(_tracker, "TH"));
 
-        _saturdayButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                if(_tracker.getSaturday()){
-                    _tracker.setSaturday(false);
-                }
-                else{
-                    _tracker.setSaturday(true);
-                }
-                Log.d("Tracker Status", _tracker.getDays());
-            }
-        });
+        _fridayButton.setOnClickListener(new ClickListenerForDaysOfTheWeek(_tracker, "FR"));
 
-        _sundayButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                if(_tracker.getSunday()){
-                    _tracker.setSunday(false);
-                }
-                else{
-                    _tracker.setSunday(true);
-                }
-                Log.d("Tracker Status", _tracker.getDays());
-            }
-        });
-    }
+        _saturdayButton.setOnClickListener(new ClickListenerForDaysOfTheWeek(_tracker, "SA"));
 
-    /**
-     * Opens a DatePickerDialog that is used to select the date of the added habit
-     */
-    private void openDatePickerDialog(){
-        DatePickerDialogFragment datePickerDialogFragment = new DatePickerDialogFragment(new DatePickerDialog.OnDateSetListener() {
-            /**
-             * Sets the text of the date select view to reflect selected date
-             * @param view
-             * @param year year of selected date
-             * @param month month of selected date (integer from 0 to 11)
-             * @param day day of month of selected date
-             */
-            @Override
-            public void onDateSet(DatePicker view, int year, int month, int day) {
-                int correctedMonth = month + 1;
-                String date =  day + "/" + correctedMonth + "/" + year;
-                _habitDateTextView.setText(date);
-            }
-        });
-        datePickerDialogFragment.show(getFragmentManager(), "DatePickerDialogFragment");
+        _sundayButton.setOnClickListener(new ClickListenerForDaysOfTheWeek(_tracker, "SU"));
```

Comment:
Wow! I like how much simpler it is, lots of lines of code saved.

Reply:
I agree!